conan/tools/google/layout.py

```diff
@@ -1,11 +1,16 @@
+import os
 
 
-def bazel_layout(conanfile, generator=None, src_folder="."):
-    """The layout for bazel is very limited, it builds in the root folder even specifying
-       "bazel --output_base=xxx" in the other hand I don't know how to inject a custom path so
-       the build can load the dependencies.bzl from the BazelDeps"""
-    conanfile.folders.build = ""
-    conanfile.folders.generators = ""
-    # used in test package for example, to know where the binaries are (editables not supported yet)
-    conanfile.cpp.build.bindirs = ["bazel-bin"]
-    conanfile.cpp.build.libdirs = ["bazel-bin"]
+def bazel_layout(conanfile, src_folder="."):
+    """Bazel layout is so limited. It does not allow to create its special symlinks in other
+    folder. See more information in https://bazel.build/remote/output-directories"""
+    subproject = conanfile.folders.subproject
+    conanfile.folders.source = src_folder if not subproject else os.path.join(subproject, src_folder)
+    conanfile.folders.build = "."  # Bazel always build the whole project in the root folder
+    # FIXME: Keeping backward-compatibility. Defaulting to "conan" in Conan 2.x.
+    conanfile.output.warning("In bazel_layout() call, generators folder changes its default value "
+                             "from './' to './conan/' in Conan 2.x")
```

Comment:
I think we can make here a explicit recomendation of setting `self.folders.generators_folder = "conan"` so that users can have a clue on what to do, but we can change that in the next PR